src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs

```diff
@@ -346,23 +346,33 @@ public Task SendStatusAsync(int statusCode)
         {
             ApplyResponseHeaders(Constants.Status200Ok);
 
-            string physicalPath = _fileInfo.PhysicalPath;
-            var sendFile = _context.Features.Get<IHttpSendFileFeature>();
-            if (sendFile != null && !string.IsNullOrEmpty(physicalPath))
-            {
-                await sendFile.SendFileAsync(physicalPath, 0, _length, _context.RequestAborted);
-                return;
-            }
-
-            Stream readStream = _fileInfo.CreateReadStream();
             try
             {
-                // Larger StreamCopyBufferSize is required because in case of FileStream readStream isn't going to be buffering
-                await StreamCopyOperation.CopyToAsync(readStream, _response.Body, _length, StreamCopyBufferSize, _context.RequestAborted);
+                string physicalPath = _fileInfo.PhysicalPath;
+                var sendFile = _context.Features.Get<IHttpSendFileFeature>();
+                if (sendFile != null && !string.IsNullOrEmpty(physicalPath))
+                {
+                    await sendFile.SendFileAsync(physicalPath, 0, _length, _context.RequestAborted);
```

Comment:
It interesting we care about passing in the client disconnected token when rendering static files in the first place. Couldn't we just ignore this by default and allow people to turn it on via a setting or something